kolibri/plugins/setup_wizard/assets/src/views/importLODUsers/LoadingTaskPage.vue

```diff
@@ -0,0 +1,191 @@
+          @click="welcomeModal = true"
+        />
+        <KButton
+          class="another-user"
+          :text="$tr('importAnother')"
+          appearance="basic-link"
+          @click="retryImport"
+        />
+      </div>
+      <KButton
+        v-else-if="loadingTask.status === 'FAILED'"
+        primary
+        :text="coreString('retryAction')"
+        @click="retryImport"
+      />
+      <span v-else></span>
+    </template>
+    <WelcomeModal
+      v-if="welcomeModal"
+      :importedFacility="facility"
+      :isLOD="true"
+      @submit="redirectToChannels"
+    />
+  </OnboardingForm>
+
+</template>
+
+
+<script>
+
+  import urls from 'kolibri.urls';
+  import redirectBrowser from 'kolibri.utils.redirectBrowser';
+  import { SessionResource } from 'kolibri.resources';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import commonSyncElements from 'kolibri.coreVue.mixins.commonSyncElements';
+  import FacilityTaskPanel from '../../../../../device/assets/src/views/FacilitiesPage/FacilityTaskPanel.vue';
+  import WelcomeModal from '../../../../../device/assets/src/views/WelcomeModal.vue';
+  import { TaskStatuses, TaskTypes } from '../../../../../device/assets/src/constants.js';
+  import OnboardingForm from '../onboarding-forms/OnboardingForm';
+  import { SetupSoUDTasksResource } from '../../api';
+
+  const welcomeDimissalKey = 'DEVICE_WELCOME_MODAL_DISMISSED';
+
+  export default {
+    name: 'LoadingTaskPage',
+    components: {
+      FacilityTaskPanel,
+      OnboardingForm,
+      WelcomeModal,
+    },
+    mixins: [commonCoreStrings, commonSyncElements],
+    data() {
+      return {
+        loadingTask: {},
+        isPolling: false,
+        welcomeModal: false,
+        user: null,
+      };
+    },
+    inject: ['lodService', 'state'],
+    computed: {
+      usersDevice() {
+        const users = this.state.value.users.filter(user => user.task === null);
+        return users;
+      },
+      facility() {
+        return this.state.value.facility;
+      },
+    },
+    beforeMount() {
+      this.isPolling = true;
+      this.pollTask();
+    },
+    methods: {
+      fullName(task_id) {
+        const user = this.state.value.users.filter(u => u.task == task_id)[0];
+        this.user = user;
+        return user.full_name;
+      },
+      pollTask() {
+        SetupSoUDTasksResource.fetchCollection({ force: true }).then(tasks => {
+          const soudTasks = tasks.filter(t => t.type === TaskTypes.SYNCLOD);
+          if (soudTasks.length > 0) {
+            this.loadingTask = {
+              ...soudTasks[0],
+              facility_name: this.facility.name,
+              full_name: this.fullName(soudTasks[0].id),
+              device_id: this.state.value.device.id,
+            };
+            if (this.loadingTask.status === TaskStatuses.COMPLETED) this.finishedTask();
+          } else this.isPolling = false;
+        });
+        if (this.isPolling) {
```

Comment:
I'm not sure of this because I'm not using `catch` in case an error  happens. If there were errors, polling would stop. I've prefered to ignore possible errors in this point.

Reply:
I think we can probably do some follow up clean up more generally, and maybe have a reusable helper resource to replace or augment `TaskResource` that can consolidate some of the polling/fetching/task initiation behaviour?

Reply:
yep, but I think that should be a reason for a different issue/PR after this is merged. That's going to be used in multiple places

Reply:
Definitely not in scope for this PR.